pkg/cli/connect_helm.go

```diff
@@ -693,3 +698,28 @@ func (cmd *connectHelm) waitForVCluster(ctx context.Context, vKubeConfig clientc
 
 	return nil
 }
+
+func checkIfAlreadyConnected(ctx context.Context, vCluster *find.VCluster) (bool, error) {
+	currentContext, _, err := find.CurrentContext()
+	if err != nil {
+		return false, err
+	}
+	if currentContext == find.VClusterContextName(vCluster.Name, vCluster.Namespace, vCluster.Context) {
+		kubeConfig, err := vCluster.ClientFactory.ClientConfig()
+		if err != nil {
+			return false, err
+		}
+
+		vKubeClient, err := kubernetes.NewForConfig(kubeConfig)
+		if err != nil {
+			return false, err
+		}
+
+		_, err = vKubeClient.CoreV1().ServiceAccounts("default").Get(ctx, "default", metav1.GetOptions{})
```

Comment:
Just a small nit and then I think we can merge this, but I guess we want to have a timeout context here, because otherwise this might hang forever